reminder/module.py

```diff
@@ -0,0 +1,409 @@
+from nextcord.ext import commands, tasks
+from nextcord.errors import HTTPException, Forbidden
+
+import dateutil.parser
+
+from pie import check, i18n, logger, utils
+from pie.utils.objects import ConfirmView
+
+from .database import ReminderStatus, ReminderItem
+
+_ = i18n.Translator("modules/reminder").translate
+bot_log = logger.Bot.logger()
+guild_log = logger.Guild.logger()
+
+
+class Reminder(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+        self.reminder.start()
+
+    def cog_unload(self):
+        self.reminder.cancel()
+
+    # LOOPS
+
+    @tasks.loop(seconds=30)
+    async def reminder(self):
+        max_remind_time = datetime.now() + timedelta(seconds=30)
+
+        items = ReminderItem.get_all(
+            status=ReminderStatus.WAITING, max_remind_date=max_remind_time
+        )
+
+        if items is not None:
+            for item in items:
+                await self._remind(item)
+
+    @reminder.before_loop
+    async def before_reminder(self):
+        print("Reminder loop waiting until ready().")
+        await self.bot.wait_until_ready()
+
+    # HELPER FUNCTIONS
+
+    async def _remind(self, item: ReminderItem):
+        reminded_user = await self._get_member(item.recipient_id, item.guild_id)
+
+        if reminded_user is None:
+            item.status = ReminderStatus.FAILED
+            item.save()
+            await bot_log.warning(
+                item.recipient_id,
+                item.guild_id,
+                "Unable to remind user - member out of bot's reach.",
+            )
+            return
+
+        utx = i18n.TranslationContext(item.guild_id, item.recipient_id)
+
+        embed = await self._get_embed(utx, item)
+
+        try:
+            message = await reminded_user.send(embed=embed)
+        except (HTTPException, Forbidden):
+            item.status = ReminderStatus.FAILED
+            item.save()
+            await bot_log.warning(
+                item.recipient_id,
+                item.guild_id,
+                "Unable to remind user - blocked PM or not enough permissions.",
+            )
+            return
+
+        item.status = ReminderStatus.REMINDED
+        item.save()
+
+        await bot_log.debug(
+            reminded_user,
+            message.channel.id,
+            "Reminder ID {id} succesfully sent to {user}".format(
+                id=item.idx, user=reminded_user.display_name
+            ),
+        )
+
+    async def _process_text(
+        self, ctx: commands.Context, datetime_str: str, text: Optional[str]
+    ):
+        if text is not None:
+            lines = ctx.message.content
+
+            lines = lines.split(datetime_str)[1]
+
+            lines = re.split(" ", lines)
+            while "" in lines:
+                lines.remove("")
+            text = " ".join(lines)
+
+            if len(text) > 1024:
+                text = text[:1024]
+                text = text[:-3] + "```" if lines.count("```") % 2 != 0 else lines
```

Comment:
What is this line doing?

Reply:
It shortens the text if it's longer than 1024 characters. And if that text has unenclosed block of code, it encloses it.

Reply:
Oh, I've got what was confusing you. Fixed.